modules/tinymce/src/core/main/ts/api/EditorUpload.ts

```diff
@@ -173,19 +149,22 @@ const EditorUpload = (editor: Editor): EditorUpload => {
           };
         });
 
-        if (filteredResult.length > 0) {
-          changeHandler.fireIfChanged();
-        }
-
         if (imagesToRemove.length > 0 && !Rtc.isRtc(editor)) {
           editor.undoManager.transact(() => {
             Arr.each(imagesToRemove, (element) => {
               editor.dom.remove(element);
+              if (blobCache.getByUri(element.src)) {
```

Comment:
`undoManager.transact` should care about dispatching `change` events on its own

Reply:
Yeah, I think we should probably just fix the test and not fire the change twice if we remove the image. The whole point of tweaking this code is to avoid change events that are not needed. So lets tweak it some more. I doubt anyone is dependent on this being emitted twice.

Reply:
ok, I removed it and fixed the test, now on error we have no more change events triggered